### PR TITLE
IGNITE-20680 Postpone destroying table data while table is reachable for RO transactions via Catalog

### PR DESCRIPTION
**Pull request by AMashenkov**
https://issues.apache.org/jira/browse/IGNITE-20680

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItCreateTableDdlTest.java [hunk 1]
```diff
@@ -340,24 +335,4 @@ public void testItIsNotPossibleToCreateTablesInSystemSchema(String schema) {
     private static Stream<Arguments> reservedSchemaNames() {
         return SYSTEM_SCHEMAS.stream().map(Arguments::of);
     }
-
-    @Disabled("https://issues.apache.org/jira/browse/IGNITE-20680")
-    @Test
-    public void concurrentDrop() {
```
[the remark this answers is not in the document]
Comment (AMashenkov) [reply modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItCreateTableDdlTest.java:346]: I'm not sure the loop is needed here.
I just can start a TX in the "past" run SELECT query after table DROP.

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RemoveIndexCommand.java [hunk 2]
```diff
@@ -58,7 +57,11 @@ private RemoveIndexCommand(int indexId) {
 
     @Override
     public List<UpdateEntry> get(Catalog catalog) {
-        CatalogIndexDescriptor index = indexOrThrow(catalog, indexId);
+        CatalogIndexDescriptor index = catalog.index(indexId);
+
+        if (index == null) {
```
[the remark this answers is not in the document]
Comment (AMashenkov) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RemoveIndexCommand.java:62]: Because the command was created as a reaction on index status change on every node and only first one could be applied and would see the index.

**Review by korlov42 — APPROVED** [review]
